### PR TITLE
Converter service

### PR DESCRIPTION
**Pull request by alexey-stankevich**

### CODE REVIEW

File: converter_service/build.gradle [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	}
+}
+
+repositories {
+	mavenCentral()
+}
+
+dependencies {
+	implementation 'org.springframework.boot:spring-boot-starter-actuator'
+	implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
```
Comment (xpinjection) [inline converter_service/build.gradle]: We don't need JPA at that stage.

File: converter_service/src/main/java/com/kyripay/converter/api/ConverterController.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+import java.util.List;
+import java.util.UUID;
+
+
+@RestController
+@Api("Converter Service")
+public class ConverterController
+{
+    @PostMapping(value = "/v1/converters/{formatId}/documents", consumes = MediaType.APPLICATION_OCTET_STREAM_VALUE)
```
Comment (xpinjection) [inline converter_service/src/main/java/com/kyripay/converter/api/ConverterController.java]: It is better to use common prefix like /api/v1 and configure it in common place like base class.

File: converter_service/src/main/java/com/kyripay/converter/dto/Document.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+@Data
+@ApiModel(value = "Document", description = "Converted payment document")
+public class Document {
+    private String id;
+    private Format format;
+    private byte[] data;
+
+
+    public Document(String id, Format format, byte[] data)
```
Comment (xpinjection) [inline converter_service/src/main/java/com/kyripay/converter/dto/Document.java]: You could use Lombok for this constructor as well.

File: converter_service/src/main/java/com/kyripay/converter/dto/Document.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        this.id = id;
+        this.format = format;
+        this.data = data;
+    }
+
+
+    public Document(byte[] data)
```
Comment (xpinjection) [inline converter_service/src/main/java/com/kyripay/converter/dto/Document.java]: Is this constructor really needed? How document may be created without format?

File: converter_service/src/main/java/com/kyripay/converter/dto/Format.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
+package com.kyripay.converter.dto;
+
+public enum Format
```
Comment (xpinjection) [inline converter_service/src/main/java/com/kyripay/converter/dto/Format.java]: Are formats not configurable and fixed?

File: converter_service/src/main/java/com/kyripay/converter/dto/Format.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  FORMAT_1("Format 1", 0, "Test format #1"),
+  FORMAT_2("Format 2", 1, "Test format #2");
+
+
+  private String name;
+  private int code;
+  private String description;
+
+
+  Format(String name, int code, String description)
```
Comment (xpinjection) [inline converter_service/src/main/java/com/kyripay/converter/dto/Format.java]: You could use Lombok for constructor and getters.

File: converter_service/src/main/java/com/kyripay/converter/dto/Document.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
+package com.kyripay.converter.dto;
+
+import io.swagger.annotations.ApiModel;
+import lombok.Data;
+
+
+@Data
+@ApiModel(value = "Document", description = "Converted payment document")
+public class Document {
```
Comment (xpinjection) [inline converter_service/src/main/java/com/kyripay/converter/dto/Document.java]: Fields are not documented for API.

File: converter_service/src/main/java/com/kyripay/converter/api/ConverterController.java (hunk 2 above)
Comment (xpinjection) [inline converter_service/src/main/java/com/kyripay/converter/api/ConverterController.java]: Confusing resource URL makes me sure I add document as a part of concrete converter (convertion rules or sample). It is better to use 'requests' or 'conversion-requests'. Taking into account async conversion procedure, as a client I need to check status of conversion and only if it succeed get URL to converted document.

File: converter_service/src/main/java/com/kyripay/converter/dto/Document.java (hunk 4 above)
Comment (alexey-stankevich) [reply converter_service/src/main/java/com/kyripay/converter/dto/Document.java]: Not needed, it's here from some intermediary step.

File: converter_service/src/main/java/com/kyripay/converter/dto/Format.java (hunk 5 above)
Comment (alexey-stankevich) [reply converter_service/src/main/java/com/kyripay/converter/dto/Format.java]: Yes, they are fixed as each format usually requires specific business logic.

**Pull request merged**